core/spec/models/spree/order_inventory_spec.rb

```diff
@@ -189,10 +193,12 @@
       end
 
       it 'only attempts to destroy as many units as are eligible, and return amount destroyed' do
-        allow(shipment).to receive_messages(inventory_units_for_item: [
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'shipped'),
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'on_hand')
-        ])
+        allow(shipment).to receive_messages(
+          inventory_units_for_item: [
+            mock_model(Spree::InventoryUnit, variant_id: variant.id, state: 'shipped', shipped?: false),
```

Comment:
@Mafi88 As this is a mocked object, we need to specify the methods which are received by this object, in this case `shipped`. `state: 'shipped'` can only be used when `#state` is called on this object to return `'shipped'` whereas `#shipped?` will return `false`.
Also, I think here it should be `shipped?: true`. Fixing this.